iothub/device/samples/PnpDeviceSamples/Thermostat/Program.cs

```diff
@@ -52,179 +67,82 @@ public static async Task Main(string[] _)
-            await s_deviceClient.SetDesiredPropertyUpdateCallbackAsync(TargetTemperatureUpdateCallbackAsync, s_deviceClient);
+            DeviceClient deviceClient;
+            switch (parameters.DeviceProvisioningType.ToLowerInvariant())
+            {
+                case "dps":
+                    s_logger.LogDebug($"Initializing via DPS");
+                    DeviceRegistrationResult dpsRegistrationResult = await ProvisionDeviceAsync(parameters, cancellationToken);
+                    var authMethod = new DeviceAuthenticationWithRegistrySymmetricKey(dpsRegistrationResult.DeviceId, parameters.DeviceSymmetricKey);
+                    deviceClient = InitializeDeviceClient(dpsRegistrationResult.AssignedHub, authMethod);
+                    break;
+
+                case "hubconnectionstring":
+                    s_logger.LogDebug($"Initializing via IoT Hub connection string");
+                    deviceClient = InitializeDeviceClient(parameters.PrimaryConnectionString);
+                    break;
+
+                default:
+                    throw new ArgumentException($"Unrecognized value for device provisioning received: {parameters.DeviceProvisioningType}." +
+                        $" It should be either \"dps\" or \"hubConnectionString\" (case-insensitive).");
+            }
+            return deviceClient;
+        }
 
-            s_logger.LogDebug($"Set handler for \"getMaxMinReport\" command.");
-            await s_deviceClient.SetMethodHandlerAsync("getMaxMinReport", HandleMaxMinReportCommandAsync, s_deviceClient);
+        // Provision a device via DPS, by sending the PnP model Id as DPS payload.
+        private static async Task<DeviceRegistrationResult> ProvisionDeviceAsync(Parameters parameters, CancellationToken cancellationToken)
+        {
+            SecurityProvider symmetricKeyProvider = new SecurityProviderSymmetricKey(parameters.DeviceId, parameters.DeviceSymmetricKey, null);
+            ProvisioningTransportHandler mqttTransportHandler = new ProvisioningTransportHandlerMqtt();
+            var pdc = ProvisioningDeviceClient.Create(parameters.DpsEndpoint, parameters.DpsIdScope, symmetricKeyProvider, mqttTransportHandler);
 
-            bool temperatureReset = true;
-            await Task.Run(async () =>
+            var pnpPayload = new ProvisioningRegistrationAdditionalData
             {
-                while (true)
-                {
-                    if (temperatureReset)
-                    {
-                        // Generate a random value between 5.0°C and 45.0°C for the current temperature reading.
-                        s_temperature = Math.Round(s_random.NextDouble() * 40.0 + 5.0, 1);
-                        temperatureReset = false;
-                    }
-
-                    await SendTemperatureAsync();
-                    await Task.Delay(5 * 1000);
-                }
-            });
+                JsonData = $"{{ \"modelId\": \"{ModelId}\" }}",
```

Comment:
When do we plan on addressing the needing to know this format and hard code?

Reply:
I'll be adding this into a pnp helper that will go into our dps assembly. This helper will be referenced in the `TemperatureController` sample.